python/ray/_private/utils.py

```diff
@@ -1190,15 +1192,45 @@ def internal_kv_put_with_retry(gcs_client, key, value, namespace, num_retries=20
     raise error
 
 
-def compute_version_info():
+@dataclass
+class VersionInfo:
+    ray_version: str
+    python_version: str
```

Comment:
Let's store the python_version as a tuple of (major, minor, patch) here and wherever we need a string, we can join at that time.